### PR TITLE
[Cu-hf5eav] Update creator view

### PR DESCRIPTION
**Pull request by Buglik**
Zrzut ekranu z 2021-04-28 o 00 03 29
Zrzut ekranu z 2021-04-28 o 00 03 37

Code is still not wrapped into components as it should be. Will do in another branch

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-hf5eav Update creator view](https://app.clickup.com/t/hf5eav)

File: web-app/src/components/tournament/creator/selectableTournamentType.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<template>
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/selectableTournamentType.vue]: The file name should be capitalized

File: web-app/src/components/tournament/creator/tournamentTypeSelector.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div class="row col-12 justify-center">
+    <selectable-tournament-type
+      v-for="item in tournamentTypesOptions"
+      :key="item.type"
+      :tournamentTypeName="item.label"
+      :tournamentType="item.type"
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/tournamentTypeSelector.vue]: try to pass only `tournamentType` and store locales as e.q. `tournament.type.single-elimination`. You will be able to get locales with `$t('tournament.type.' + tournamentType)`

in locales files:
"single-elimination": "txt"

File: web-app/src/pages/tournament/TournamentCreator.vue [hunk 3]
```diff
@@ -1,134 +1,169 @@
...
-              v-model="tournamentName"
-              :validate="val => val.length > 3 && val.length < 40"
+        <!-- <q-stepper-navigation>
+          <q-btn
+            :disable="!activeTournamentType.length"
+            @click="step = 2"
+            color="primary"
+            label="Continue"
+          />
+        </q-stepper-navigation> -->
```
Comment (Zummek) [inline web-app/src/pages/tournament/TournamentCreator.vue]: ?

File: web-app/src/components/tournament/creator/selectableTournamentType.vue [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    </q-btn>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import EventBus from '../../../services/EventBus';
+
+@Component
+export default class SelectableTournamentType extends Vue {
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/selectableTournamentType.vue]: I suggest to rename this class to `TournamentTypeSelectorItem`

File: web-app/src/components/tournament/creator/selectableTournamentType.vue [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div class="row col-8 col-sm-4 justify-center q-ma-md">
+    <q-btn class="row col-12" no-caps padding="none">
+      <q-card
+        flat
+        class="col-12 tournamentTypeCard"
+        @click="updateTournamentType"
+      >
+        <q-icon class="col-12" :name="iconName" style="font-size: 1000%" />
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/selectableTournamentType.vue]: `font-size: 1000%` 
LoL
okej xd

File: web-app/src/components/tournament/creator/selectableTournamentType.vue [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import EventBus from '../../../services/EventBus';
+
+@Component
+export default class SelectableTournamentType extends Vue {
+  @Prop({ type: String, required: true }) readonly tournamentTypeName!: string;
+  @Prop({ type: String, required: true }) readonly tournamentType!: string;
+  @Prop({ type: String, required: true }) readonly iconName!: string;
+
+  private updateTournamentType() {
+    EventBus.$emit('updateActiveTournamentType', this.tournamentType);
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/selectableTournamentType.vue]: I wonder if this is a good use of the EventBus. You can emit this event to parent component and that parent can forward emit to parent of the parent 
As name of this event I suggest to use `select` like in selectors

You will be able to receive this as e.q. 
```javascript
<component
   @myCustomEvent="forwardEvent"
/>

...

<script>
...
forwardEvent(data) {this.$emit('myCustomEvent', data}
```

It's not every important but would be nice to code this as above, maybe in correction task that You mentioned in the description

**Review by Zummek — APPROVED** [review]
Take a look at the comments, make corrections, and you'll be fine. I will review templates/html in next mentioned task

File: web-app/src/pages/tournament/TournamentCreator.vue (hunk 3 above)
Comment (Buglik) [reply web-app/src/pages/tournament/TournamentCreator.vue]: trash

File: web-app/src/components/tournament/creator/selectableTournamentType.vue (hunk 6 above)
Comment (Buglik) [reply web-app/src/components/tournament/creator/selectableTournamentType.vue]: done so after consulting with @eastemployeer 
but probably will do in the mentioned task

File: web-app/src/pages/tournament/TournamentCreator.vue [hunk 7]
```diff
@@ -260,6 +249,13 @@ export default class TournamentCreator extends Vue {
   }
 
   private validation() {
+    if (
+      this.activeTournamentType === 'single-elimination' ||
+      this.activeTournamentType === 'round-robin'
+    )
+      return true;
+    else return false;
+
```
Comment (Zummek) [inline web-app/src/pages/tournament/TournamentCreator.vue]: I'm not sure but You  wanna check data below.?So You cant return here true. Validation fun should check all if's and return false on first not valid object. On the end of the function should be `return true` to return true when all objects are valid

**Review by Zummek — APPROVED** [review]
LGTM

File: web-app/src/pages/tournament/TournamentCreator.vue (hunk 7 above)
Comment (Buglik) [reply web-app/src/pages/tournament/TournamentCreator.vue]: true af
correcting right away

**Pull request merged**